### PR TITLE
Add support for IRCv3 Client Capability Negotiation

### PR DESCRIPTION
**Pull request by hloeung**
This is for https://github.com/42wim/matterircd/issues/476

Testing using irssi:

```
INFO[2022-10-12T19:43:04+11:00] New connection: 127.0.0.1:54024               module=matterircd
DEBU[2022-10-12T19:43:04+11:00] using paste buffer timeout: 1000              module=matterircd
DEBU[2022-10-12T19:43:04+11:00] <- CAP LS 302                                 module=matterircd
DEBU[2022-10-12T19:43:04+11:00] <- JOIN :                                     module=matterircd
DEBU[2022-10-12T19:43:04+11:00] -> :matterircd CAP * LS :                     module=matterircd
DEBU[2022-10-12T19:43:04+11:00] <- CAP END                                    module=matterircd
DEBU[2022-10-12T19:43:04+11:00] <- NICK hloeung                               module=matterircd
```

Testing with netcat:

```
$ nc -C -v 127.0.0.1 6697
Connection to 127.0.0.1 6697 port [tcp/ircs-u] succeeded!
CAP LS 302
:matterircd CAP * LS :
CAP DDDD
:matterircd 410 * DDDD :Invalid or unsupported CAP command
```

### CODE REVIEW

File: mm-go-irckit/server.go [hunk 1]
```diff
@@ -444,6 +444,21 @@ outerloop:
 				u.Pass = msg.Params
 			case irc.JOIN:
 				s.EncodeMessage(u, irc.ERR_NOTREGISTERED, []string{"*"}, "Please register first")
+			// https://ircv3.net/specs/extensions/capability-negotiation.html
+			case irc.CAP:
+				subcommand := msg.Params[0]
+				switch subcommand {
+				case irc.CAP_LS, irc.CAP_LIST:
+					params := fmt.Sprintf("* %s :", subcommand)
+					s.EncodeMessage(u, irc.CAP, []string{params}, "")
```
Comment (hloeung) [inline mm-go-irckit/server.go:453]: @42wim not sure whether to `//nolint:errcheck` these to silence the golangci-lint failures or actually check and discard the return values. Both of which would make it inconsistent with the rest of the s.EncodeMessage() calls within this file.

Comment (42wim) [reply mm-go-irckit/server.go:453]: fine to //nolint it

Comment (42wim) [reply mm-go-irckit/server.go]: you need to set //nolint:errcheck because of changes in go1.19 wrt gofmt

Comment (hloeung) [reply mm-go-irckit/server.go]: Pushed up a fix, thanks!

**Comment by 42wim** [comment]
Thanks for your work on this!

**Comment by hloeung** [comment]
Thanks for reviewing!

**Pull request merged**